src/ImageSharp/IO/BufferedReadStream.cs

```diff
@@ -186,7 +172,7 @@ public override int Read(Span<byte> buffer)
 
         // Too big for remaining buffer but less than entire buffer length
         // Copy to buffer then read from there.
-        if (count + this.readBufferIndex > this.BufferSize)
+        if (this.readBufferIndex < 0 || (count + this.readBufferIndex > this.BufferSize))
```

Comment:
The `+` could overflow (at least in theory).
```suggestion
        if (this.readBufferIndex < 0 || (this.readBufferIndex > this.BufferSize - count))
```
Or to remove one check
```suggestion
        if ((uint)this.readBufferIndex > (uint)(this.BufferSize - count))
```

Reply:
Should have committed the second one. Will do so later. Thanks!